### ISSUE TITLE
Some keys do not work in certain situations.

### ISSUE THREAD

**Issue by ncfw** [fcitx/fcitx5-hangul#13]
**Some keys do not work in certain situations.**

I'm using Google Spreadsheets.
When I select a cell and make my first input while holding down the Shift key, some keys are not entered.

Once I enter the cell, I can enter data.
(Place the cursor on the cell and press the Enter key or double-click the left mouse button)

But, some keys are not entered before entering the cell.

The symbol keys from 6(^), Y, H, B to the right end, and also the `(~) key cannot be input.

**Conditions of occurrence**

· Using “**Sebeolsik** Final” (or 390) Korean Keyboard Layout.  (works fine with Dubeolsik) 

· **Wayland** (works fine in X11 mode)

· Firefox, Chrome (**Wayland Native**)
(For Chrome, I used the options --ozone-platform-hint=auto / wayland, and --enable-wayland-ime --wayland-text-input-version=3.)

· In Google Spreadsheets, select a cell by clicking once or moving the cursor. (Do not double-click or 
press Enter to enter.)

· **In Korean input mode**, **hold down the Shift key** and type the keys shown below.

~
  ^ , & , * , ( , ) , _ , +
  Y , U , I , O , P , { , } , |
   H , J , K , L , : , “
   B , N , M , < , > , ? 

· Other keys work fine even when Shift is pressed.

Could you please look into this matter?

p.s.
I also tested it with ibus. 
(ubuntu 24.04 gnome wayland ibus )
Chrome (Wayland Native) : Working fine
Firefox: Same problem occurs

**Comment by wengxt** [comment]
I tried to test ibus, but get the same result with fcitx. You might be running chrome under a different mode when you're testing with ibus.

I don't think there's anything can be done on fcitx side. The common thing with the key you listed is that, they produce text directly without preedit. I believe firefox uses preedit to detect ime and that works better.

But, if you just want to use fcitx + chrome + native wayland + GNOME at this moment:

This shall work
```
GTK_IM_MODULE=fcitx chromium --gtk-version=4 --ozone-platform=wayland 
```

**Comment by ncfw** [comment]
Thanks for the testing, and your comment.

(1) with ibus
Since you said you got the same results as fcitx with ibus, I'll explain the method I tested in more detail.

I used the xlsclients command to see if it was running Wayland natively.
When I boot into Ubuntu and run the xlsclients command without doing anything, I get the following three things.

 < gsd-xsetting , ibus-x11 , mutter-x11-frames >

When I click on the Google Chrome browser icon to launch it and then use the xlsclients command, four things are displayed like this.

< gsd-xsetting , ibus-x11 , google-chrome-stable , mutter-x11-frames >


If it is run in xwayland mode, it will be displayed in the list like this.

I used the options below to run Google Chrome as Wayland native.

`$ google-chrome-stable --ozone-platform=wayland --gtk-version=4
`

After running Google Chrome with the options listed above, I ran the xlsclients command and confirmed that "google-chrome-stable" was not added.

So, I thought it was running in Wayland native mode.

And when I pressed those keys while holding down the shift key, they were entered without any problem.



(2) with fcitx5 (arch linux , gnome, wayland )

`GTK_IM_MODULE=fcitx google-chrome-stable --gtk-version=4 --ozone-platform=wayland 
`

When running with the above options, the Korean input method did not work at all.

I registered GTK_IM_MODULE=fcitx in /etc/environment.

`$ google-chrome-stable --ozone-platform-hint=auto --enable-wayland-ime --wayland-text-input-version=3
`

※ The above options are referenced from ArchWiki.
https://wiki.archlinux.org/title/Chromium

Except for the problem I wrote in the first post, Korean input works fine.

When I ran xlsclients, "google-chrome-stable" was not listed either.



(3)

On the sebeolsik-final keyboard, those keys are numbers and symbols.
That means I can't start cell input with numbers and symbols.

If you have any idea where I should talk about this problem, please let me know.

Thank you.

**Comment by ncfw** [comment]
Thank yoiu for your reply.

It was just a simple typing mistake on my part. 

I typed --gtk-version4 instead of --gtk-version=4.



$ GTK_IM_MODULE=fcitx google-chrome-stable --gtk-version=4 --ozone-platform-wayland 

It works. Korean input works well. It works well in Google Spreadsheets!



However, I already declared GTK_IM_MODULE=fcitx in /etc/environment.

I was wondering if it was really necessary to set it up separately.

so, I tried
 
$ google-chrome-stable --gtk-version=4 --ozone-platform-wayland

I works too!



I can't believe it.
I originally used ibus and used the --gtk-version=4 option.

I deleted ibus and installed fcitx5, and at that time, when I ran it with the --gtk-version=4 option, Korean characters were not input at all.

So I looked for information on the Arch Linux Wiki, and started using the --enable-wayland-ime --wayland-text-input-version=3 option.



I guess I did something wrong, like testing before changing GTK_IM_MODULE= in /etc/environment from ibus to fcitx.


Ultimately, it seems like the problem is caused by using --wayland-text-input-version=3 as wengxt pointed out.


Thanks to wengxt, I can now use Google Spreadsheets smoothly with Chrome browser. Thank you.


p.s. 
Firefox is still having issues, but there doesn't seem to be anything I can do about it.
However,  I found out that this is not an fcitx issue.

Thanks again wengxt.
